plugins/modules/foreman_search_facts.py

```diff
@@ -107,16 +124,18 @@ def main():
             resource=dict(type='str', required=True),
             search=dict(default=""),
             full_details=dict(type='bool', aliases=['info'], default='false'),
+            params=dict(type='dict'),
         ),
     )
 
     module_params = module.clean_params()
     resource = module_params['resource']
     search = module_params['search']
+    params = module_params.get('params')
```

Comment:
```suggestion
    params = module_params.get('params').copy()
```
The `module_params` are a filtered copy of ansibles internal data, but it is only a shallow copy. So this command retrieves another reference to internal data.

Reply:
I have tried similar. The above-described behavior happens even when I hard-code the call as `response = module.list_resource(resource, search, {"organization_id": "1"} )` and when I'm not touching any params

Reply:
I think it is the call in:
https://github.com/theforeman/foreman-ansible-modules/pull/537/files#diff-0424e051e61216ca71bbdfedba26401bR138

Could you try again with a copy of params?

Reply:
For completeness sake:

```
    "error": {
        "message": "Association not found for organization"
    }, 
    "invocation": {
        "module_args": {
            "full_details": false, 
            "organization": "org_RH", 
            "params": {}, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "resource": "settings", 
            "search": "", 
            "server_url": "https://dh2.rh.local", 
            "username": "admin", 
            "validate_certs": false
        }
    }, 

```

Reply:
See, the strangeness in module_args['params'] is gone, where it has been in your previous example:
```
"params": {
                "organization_id": 1, 
                "per_page": 4294967296, 
                "search": ""
            }
```

Reply:
can you try

```python
params = module.foremanapi.resource(resource).action('index').prepare_params(params)
```

after the `if organization` block? that should filter out params that the resource can't understand.